webknossos/webknossos/geometry/mag.py

```diff
@@ -1,87 +1,112 @@
-        elif isinstance(mag, Mag):
-            self.mag = mag.mag
-        elif isinstance(mag, np.ndarray):
-            assert mag.shape == (3,)
-            self.mag = list(mag)
-
-        if self.mag is None or len(self.mag) != 3:
-            raise ValueError(
-                "magnification must be int or a vector3 of ints or a string shaped like e.g. 2-2-1"
-            )
-
-        for m in self.mag:
-            assert log2(m) % 1 == 0, "magnification needs to be power of 2."
+@attr.frozen
+class Mag:
+    _mag: Vec3Int = attr.ib(converter=import_mag)
+
+    @property
+    def x(self) -> int:
+        return self._mag.x
+
+    @property
+    def y(self) -> int:
+        return self._mag.y
+
+    @property
+    def z(self) -> int:
+        return self._mag.z
+
+    @property
+    def max_dim(self) -> int:
+        return max(self._mag)
 
     def __lt__(self, other: Any) -> bool:
-        return max(self.mag) < (max(Mag(other).to_array()))
+        return self.max_dim < Mag(other).max_dim
 
     def __le__(self, other: Any) -> bool:
-        return max(self.mag) <= (max(Mag(other).to_array()))
+        return self.max_dim <= Mag(other).max_dim
 
     def __eq__(self, other: Any) -> bool:
-        return all(m1 == m2 for m1, m2 in zip(self.mag, Mag(other).mag))
+        return self.to_vec3_int() == Mag(other).to_vec3_int()
 
     def __str__(self) -> str:
         return self.to_layer_name()
 
-    def __expr__(self) -> str:
+    def __repr__(self) -> str:
         return f"Mag({self.to_layer_name()})"
 
     def to_layer_name(self) -> str:
-        x, y, z = self.mag
+        x, y, z = self._mag
         if x == y and y == z:
             return str(x)
         else:
             return self.to_long_layer_name()
 
     def to_long_layer_name(self) -> str:
-        x, y, z = self.mag
+        x, y, z = self._mag
         return "{}-{}-{}".format(x, y, z)
 
-    def to_array(self) -> List[int]:
-        return self.mag
+    def to_list(self) -> List[int]:
+        return self._mag.to_list()
 
-    def scaled_by(self, factor: int) -> "Mag":
-        return Mag([mag * factor for mag in self.mag])
+    def to_np(self) -> np.ndarray:
+        return self._mag.to_np()
 
-    def scale_by(self, factor: int) -> None:
-        self.mag = [mag * factor for mag in self.mag]
+    def to_vec3_int(self) -> Vec3Int:
+        return self._mag
 
-    def divided(self, coord: List[int]) -> List[int]:
-        return [c // m for c, m in zip(coord, self.mag)]
+    def to_tuple(self) -> Tuple[int, int, int]:
+        return self._mag.to_tuple()
 
-    def divide_by(self, d: int) -> None:
-        self.mag = [mag // d for mag in self.mag]
+    def scaled_by(self, factor: int) -> "Mag":
+        return Mag([mag * factor for mag in self._mag])
 
-    def divided_by(self, d: int) -> "Mag":
-        return Mag([mag // d for mag in self.mag])
+    def divided(self, coord: List[int]) -> List[int]:
+        return [c // m for c, m in zip(coord, self._mag)]
```

Comment:
For such arithmetics we might also have Mag inherit from Vec3Int directly, but I'm not really sure if this is helpful. Let's defer this discussion to another time ;-)

Reply:
Yeah I am unsure about that as well. Let’s defer